src/lib/WIFI/devWIFI.cpp

```diff
@@ -796,6 +801,238 @@ static void WebUploadForceUpdateHandler(AsyncWebServerRequest *request) {
   }
 }
 
+#if defined(RADIO_LR1121)
+static size_t expectedFilesize;
+static SX12XX_Radio_Number_t updatingRadio;
+extern LR1121Hal hal;
+static uint8_t packet[256];
+static size_t left_over;
+
+static void writeLR1121Bytes(uint8_t *data, uint32_t sector_size) {
+    uint8_t header[6 + 256] = {
+        (uint8_t)(LR11XX_BL_WRITE_FLASH_ENCRYPTED_OC >> 8),
+        (uint8_t)(LR11XX_BL_WRITE_FLASH_ENCRYPTED_OC),
+        (uint8_t)(totalSize >> 24),
+        (uint8_t)(totalSize >> 16),
+        (uint8_t)(totalSize >> 8),
+        (uint8_t)(totalSize)
+    };
+    if (left_over)
+    {
+        DBGLN("left %x", left_over);
+        memcpy(header + 6, packet, left_over);
+    }
+    if (data != NULL)
+    {
+        DBGLN("new %x", sector_size - left_over);
+        memcpy(header + 6 + left_over, data, sector_size - left_over);
+    }
+    DBGLN("flashing %x at %x", sector_size, totalSize);
+
+    // Have to do this the OLD way, so we can pump out more than 64 bytes in one message
+    digitalWrite(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_NSS : GPIO_PIN_NSS_2, LOW);
+    SPIEx.transferBytes(header, NULL, 6 + sector_size);
+    digitalWrite(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_NSS : GPIO_PIN_NSS_2, HIGH);
+
+    while (digitalRead(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_BUSY : GPIO_PIN_BUSY_2) == HIGH)
+    {
+        delay(1);
+    }
+    DBGLN("flashed");
+}
+
+static void WebUploadLR1121ResponseHandler(AsyncWebServerRequest *request) {
+    // Complete upload and set error flag
+    bool uploadError = false;
+    uint8_t param = 0;
+    writeLR1121Bytes(NULL, left_over);
+    totalSize += left_over;
+
+    SPIEx.setHwCs(true);
+    if (GPIO_PIN_NSS_2 != UNDEF_PIN)
+    {
+        spiAttachSS(SPIEx.bus(), 1, GPIO_PIN_NSS_2);
+    }
+
+    if (totalSize == expectedFilesize)
+    {
+        DBGLN("reboot 1121");
+        uint8_t reboot_cmd[] = {
+            (uint8_t)(LR11XX_BL_REBOOT_OC >> 8),
+            (uint8_t)LR11XX_BL_REBOOT_OC,
+            0
+        };
+        SPIEx.write(updatingRadio, reboot_cmd, 3);
+        while(!hal.WaitOnBusy(updatingRadio));
+
+        DBGLN("check not in BL mode");
+        packet[0] = LR11XX_SYSTEM_GET_VERSION_OC >> 8;
```

Comment:
I feel like reusing the packet buffer here is not a great idea since that's for storing left over data bytes and this confuses the usage for no benefit. Just use a local small stack buffer for all of these so that the reader doesn't have to wonder about the data going into packet being used somewhere else?